### PR TITLE
fix: encode uri for delete, restore and favorite

### PR DESCRIPTION
**Pull request by kesselb**
## Summary

To encode # and other characters properly

Similar to https://github.com/nextcloud/server/pull/39842 but different ;)

`#` is a valid character for our filenames but also for uris in general and therefore not encoded by default. 
To use `#` in filenames we need to encode it.

## TODO

- [ ] CI
- [ ] Review

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by kesselb** [comment]
Version for stable27: https://github.com/nextcloud/server/pull/40642

**Review by ChristophWurst — APPROVED** [review]
Makes sense

**Review by Pytal — APPROVED** [review]

**Comment by ChristophWurst** [comment]
CI failure is unrelated to the code changes

**Comment by skjnldsv** [comment]
UI is broken and show lots of encoded files now :)
Reverting this in followup. The data model should NOT include encoded string, it should not be request-aware. A Node object contains raw data and should never be transformed/oriented towards a specific usage (here a request context for example)

File: apps/files_trashbin/src/services/trashbin.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ const resultToNode = function(node: FileStat): File | Folder {
 
 	const nodeData = {
 		id: node.props?.fileid as number || 0,
-		source: generateRemoteUrl('dav' + rootPath + node.filename),
+		source: generateRemoteUrl(encodePath('dav' + rootPath + node.filename)),
```
Comment (skjnldsv) [inline apps/files_trashbin/src/services/trashbin.ts:49]: `encodePath` is no even defined here...

**Pull request merged**